code/game/objects/items/weapons/RCD.dm

```diff
@@ -21,76 +48,123 @@ RCD
-		/obj/machinery/door/airlock/medical = "Medical", /obj/machinery/door/airlock/medical/glass = "Medical (Glass)",
-		/obj/machinery/door/airlock/maintenance = "Maintenance", /obj/machinery/door/airlock/maintenance/glass = "Maintenance (Glass)",
-		/obj/machinery/door/airlock/external = "External", /obj/machinery/door/airlock/external/glass = "External (Glass)",
-		/obj/machinery/door/airlock/maintenance/external = "External Maintenance", /obj/machinery/door/airlock/maintenance/external/glass = "External Maintenance (Glass)",
-		/obj/machinery/door/airlock/freezer = "Freezer",
-		/obj/machinery/door/airlock/mining = "Mining", /obj/machinery/door/airlock/mining/glass = "Mining (Glass)",
-		/obj/machinery/door/airlock/research = "Research", /obj/machinery/door/airlock/research/glass = "Research (Glass)",
-		/obj/machinery/door/airlock/atmos = "Atmospherics", /obj/machinery/door/airlock/atmos/glass = "Atmospherics (Glass)",
-		/obj/machinery/door/airlock/science = "Science", /obj/machinery/door/airlock/science/glass = "Science (Glass)",
-		/obj/machinery/door/airlock/hatch = "Airtight Hatch",
-		/obj/machinery/door/airlock/maintenance_hatch = "Maintenance Hatch")
-
-/obj/item/rcd/New()
-	desc = "A RCD. It currently holds [matter]/[max_matter] matter-units."
+
+	// Stupid shit
+	var/static/allowed_targets = list(/turf, /obj/structure/grille, /obj/structure/window, /obj/structure/lattice, /obj/machinery/door/airlock)
+
+/obj/item/rcd/Initialize()
+	. = ..()
 	spark_system = new /datum/effect_system/spark_spread
 	spark_system.set_up(5, 0, src)
 	spark_system.attach(src)
 	GLOB.rcd_list += src
+
 	door_accesses_list = list()
 	for(var/access in get_all_accesses())
 		door_accesses_list[++door_accesses_list.len] = list("name" = get_access_desc(access), "id" = access, "enabled" = (access in door_accesses))
-	return
+
+/obj/item/rcd/examine(mob/user)
+	. = ..()
+	to_chat(user, "MATTER: [matter]/[max_matter] matter-units.")
+	to_chat(user, "MODE: [mode].")
 
 /obj/item/rcd/Destroy()
 	QDEL_NULL(spark_system)
 	GLOB.rcd_list -= src
 	return ..()
 
+/obj/item/rcd/proc/get_airlock_image(airlock_type)
+	var/obj/machinery/door/airlock/proto = airlock_type
+	var/ic = initial(proto.icon)
+	var/mutable_appearance/MA = mutable_appearance(ic, "closed")
+	if(!initial(proto.glass))
+		MA.overlays += "fill_closed"
+	// Not scaling these down to button size because they look horrible then, instead just bumping up radius.
+	return MA
+
+/obj/item/rcd/proc/check_menu(mob/living/user)
+	if(!istype(user))
+		return FALSE
+	if(user.incapacitated() || !user.Adjacent(src))
+		return FALSE
+	return TRUE
+
 /obj/item/rcd/attackby(obj/item/W, mob/user, params)
 	..()
+
 	if(istype(W, /obj/item/rcd_ammo))
 		var/obj/item/rcd_ammo/R = W
 		if((matter + R.ammoamt) > max_matter)
 			to_chat(user, "<span class='notice'>The RCD can't hold any more matter-units.</span>")
 			return
 		matter += R.ammoamt
-		user.drop_item()
-		qdel(W)
+		if(!user.unEquip(R))

```

Comment:
Has to be done before the matter adding. Now you get the matter and it'll stay in your hand